### PR TITLE
Fix big endian issue in Interlocked test cases

### PR DESCRIPTION
**Pull request by giritrivedi**
Fix big indian issue wiht BitConverted as part of intercloked test cases. 
These were causing failures in runtime_community pipeline

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/Threading/Interlocked.cs [hunk 1]
```diff
@@ -308,7 +308,7 @@ public static unsafe ushort CompareExchange(ref ushort location1, ushort value,
             nuint offset = Unsafe.OpportunisticMisalignment(ref location1, sizeof(uint));
             ref uint alignedRef = ref Unsafe.As<ushort, uint>(ref Unsafe.SubtractByteOffset(ref location1, offset));
             int bitOffset =
-                (int)((BitConverter.IsLittleEndian ? offset : sizeof(uint) - offset - sizeof(byte)) * 8); // to bit offset
+                (int)((BitConverter.IsLittleEndian ? offset : sizeof(uint) - offset - sizeof(short)) * 8); // to bit offset
```
[the remark this answers is not in the document]
Comment (giritrivedi) [reply src/libraries/System.Private.CoreLib/src/System/Threading/Interlocked.cs]: Thanks for the quick review.

**Comment by giritrivedi** [comment]
@akoeplinger , can you run "runtime community" pipeline on this. Lets see if the fix holds good for System.Threads.tests failures seen as part of https://github.com/dotnet/runtime/issues/101380

**Comment by giritrivedi** [comment]
@akoeplinger Thanks for initiating the run. 
The run CI looks green. 
Do you mind enabling the disabled test cases which were causing hang and check if the fix still holds good ?

**Review by akoeplinger — APPROVED** [review]

**Comment by akoeplinger** [comment]
/ba-g failures are not related, this change only affects big endian platforms

**Comment by akoeplinger** [comment]
Yep, I'm doing it in https://github.com/dotnet/runtime/pull/101527 so that we don't need to restart CI if it doesn't.

**Pull request merged**